src/pub.ts

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+import * as channels from "./channels";
+import * as child_process from "child_process";
+import * as os from "os";
+import * as path from "path";
+import * as project from "./project";
+import * as vs from "vscode";
+
+export class PubManager {
+	private sdk: string;
+
+	constructor(sdk: string) {
+		this.sdk = sdk;
+	}
+
+	registerCommands(context: vs.ExtensionContext) {
+		context.subscriptions.push(vs.commands.registerCommand("pub.get", selection => {
+			this.runPub("get", selection);
+		}));
+		context.subscriptions.push(vs.commands.registerCommand("pub.upgrade", selection => {
+			this.runPub("upgrade", selection);
+		}));
+	}
+
+	private runPub(command: String, selection?) {
+		let root = vs.workspace.rootPath;
+		let projectPath = selection
+			? path.dirname(selection.path)
```

Comment:
I've had to change this from `.path` to `.fsPath` since `path` is for urls and on Windows returns things like `m:/work/` which then got prefixed with `c:\` and it all went bad. I believe it'll still work on Mac/Linux this way (the Code codebase has lots of instances of `fsPath` for stuff like this) but it'd be good to confirm.